From ca56674d42dd8ebed0c6e5050f45c7c5fb54c57b Mon Sep 17 00:00:00 2001
Message-Id: <ca56674d42dd8ebed0c6e5050f45c7c5fb54c57b.1431284209.git.chang-joon.lee@intel.com>
In-Reply-To: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
References: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
From: Michel Thierry <michel.thierry@intel.com>
Date: Thu, 2 Apr 2015 12:24:09 +0100
Subject: [PATCH 30/68] FROM_UPSTREAM [VPG]: drm/i915: Use complete address
 space in true PPGTT

True PPGTT is capable of having a full address space, even if the system
has less allocated memory.

Note that aliasing PPGTT always aliases the GGTT and thus should remain
of the same size.

Change-Id: I790ade4f6dd953bb4d044da0ffe37c7c0d5101cd
Upstream-repo: https://www.mail-archive.com/intel-gfx@lists.freedesktop.org/msg57034.html
Upstream-commit: none
Upstream-as-of-tag: none
Issue: GMINL-5449
Issue: VIZ-4619
Signed-off-by: Michel Thierry <michel.thierry@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_gtt.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_gtt.c b/drivers/gpu/drm/i915/i915_gem_gtt.c
index 5a661cd..45a78b3 100644
--- a/drivers/gpu/drm/i915/i915_gem_gtt.c
+++ b/drivers/gpu/drm/i915/i915_gem_gtt.c
@@ -1024,11 +1024,9 @@ static int gen8_aliasing_ppgtt_init(struct i915_hw_ppgtt *ppgtt)
 
 static int gen8_ppgtt_init(struct i915_hw_ppgtt *ppgtt)
 {
-	struct drm_device *dev = ppgtt->base.dev;
-	struct drm_i915_private *dev_priv = dev->dev_private;
 	int ret;
 
-	ret = gen8_ppgtt_init_common(ppgtt, dev_priv->gtt.base.total);
+	ret = gen8_ppgtt_init_common(ppgtt, (1ULL << 32));
 	if (ret)
 		return ret;
 
-- 
1.7.9.5

